web/components/ResourceWatcher.vue

```diff
@@ -0,0 +1,215 @@
+            if (done) {
+              snackbarstore.setServerErrorMessage(
+                "The watch stream is terminated. Please reload your browser if you don't expect this."
+              );
+              return;
+            }
+            buffer += utf8Decoder.decode(value);
+            buffer = onNewLine(buffer, async (chunk: string) => {
+              if (chunk.trim().length === 0) {
+                return;
+              }
+              try {
+                const event = JSON.parse(chunk) as WatchEvent;
+                switch (event.Kind) {
+                  case resourceKind.PODS: {
+                    pstore.watchEventHandler(
+                      event.EventType,
+                      event.Obj as V1Pod
+                    );
+                    pstore.setLastResourceVersion(event.Obj as V1Pod);
+                    break;
+                  }
+                  case resourceKind.NODES: {
+                    nstore.watchEventHandler(
+                      event.EventType,
+                      event.Obj as V1Node
+                    );
+                    nstore.setLastResourceVersion(event.Obj as V1Node);
+                    break;
+                  }
+                  case resourceKind.PVS: {
+                    pvstore.watchEventHandler(
+                      event.EventType,
+                      event.Obj as V1PersistentVolume
+                    );
+                    pvstore.setLastResourceVersion(
+                      event.Obj as V1PersistentVolume
+                    );
+                    break;
+                  }
+                  case resourceKind.PVCS: {
+                    pvcstore.watchEventHandler(
+                      event.EventType,
+                      event.Obj as V1PersistentVolumeClaim
+                    );
+                    pvcstore.setLastResourceVersion(
+                      event.Obj as V1PersistentVolumeClaim
+                    );
+                    break;
+                  }
+                  case resourceKind.SCS: {
+                    storageclassstore.watchEventHandler(
+                      event.EventType,
+                      event.Obj as V1StorageClass
+                    );
+                    storageclassstore.setLastResourceVersion(
+                      event.Obj as V1StorageClass
+                    );
+                    break;
+                  }
+                  case resourceKind.PCS: {
+                    priorityclassstore.watchEventHandler(
+                      event.EventType,
+                      event.Obj as V1PriorityClass
+                    );
+                    priorityclassstore.setLastResourceVersion(
+                      event.Obj as V1PriorityClass
+                    );
+                    break;
+                  }
+                }
+              } catch (error) {
+                console.log("Error while parsing", chunk, "\n", error);
```

Comment:
better to let users know the error happens via `snackbarstore.setServerErrorMessage`? wdyt?

Reply:
+1
Note, Removed the `chunk`. Because its information is not for any user but for the developer of the simulator, I think.